hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/security/token/delegation/AbstractDelegationTokenSecretManager.java

```diff
@@ -531,9 +557,9 @@ protected DelegationTokenInformation checkToken(TokenIdent identifier)
     }
     return info;
   }
-  
+
   @Override
-  public synchronized byte[] retrievePassword(TokenIdent identifier)
+  public byte[] retrievePassword(TokenIdent identifier)
```

Comment:
Why remove synchronized?

Reply:
it is not required because currentTokens is a concurrenthashmap so satisfies the happens before memory visibility requirements for this method.